crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -0,0 +1,318 @@
+    /// #     type_registry: TypeRegistry,
+    /// # }
+    /// // A resource that can hold any component that implements reflect as a boxed reflect component
+    /// #[derive(Resource)]
+    /// struct SpecialComponentHolder{
+    ///     component: Box<dyn Reflect>,
+    /// }
+    /// #[derive(Component, Reflect, FromReflect, Default)]
+    /// #[reflect(Component)]
+    /// struct ComponentA(u32);
+    /// #[derive(Component, Reflect, FromReflect, Default)]
+    /// #[reflect(Component)]
+    /// struct ComponentB(String);
+    ///
+    /// fn insert_reflected_component(
+    ///     mut commands: Commands,
+    ///     mut type_registry: ResMut<TypeRegistryResource>,
+    ///     mut special_component_holder: ResMut<SpecialComponentHolder>
+    ///     ) {
+    ///     #
+    ///     # type_registry.type_registry.register::<ComponentA>();
+    ///     # let mut registration = type_registry
+    ///     #     .type_registry
+    ///     #     .get_mut(std::any::TypeId::of::<ComponentA>())
+    ///     #     .unwrap();
+    ///     # registration.insert(<ReflectComponent as FromType<ComponentA>>::from_type());
+    ///     #
+    ///     # type_registry.type_registry.register::<ComponentB>();
+    ///     # let mut registration = type_registry
+    ///     #     .type_registry
+    ///     #     .get_mut(std::any::TypeId::of::<ComponentB>())
+    ///     #     .unwrap();
+    ///     # registration.insert(<ReflectComponent as FromType<ComponentB>>::from_type());
+    ///     // Create a set of new boxed reflect components to use
+    ///     let boxed_reflect_component_a = Box::new(ComponentA(916)) as Box<dyn Reflect>;
+    ///     let boxed_reflect_component_b = Box::new(ComponentB("NineSixteen".to_string())) as Box<dyn Reflect>;
+    ///
+    ///     // You can overwrite the component in the resource with either ComponentA or ComponentB
+    ///     special_component_holder.component = boxed_reflect_component_a;
+    ///     special_component_holder.component = boxed_reflect_component_b;
+    ///     
+    ///     // No matter which component is in the resource and without knowing the exact type, you can
+    ///     // use the insert_reflected entity command to insert that component into an entity.
+    ///     commands
+    ///         .spawn_empty()
+    ///         .insert_reflected(special_component_holder.component.clone_value(), type_registry.type_registry.clone());
+    /// }
+    ///
+    /// ```
+    fn insert_reflected(
+        &mut self,
+        component: Box<dyn Reflect>,
+        type_registry: TypeRegistry,
+    ) -> &mut Self;
+
+    /// Removes the component of the same type as the supplied boxed reflect component from the entity
+    /// using the reflection data in the supplied [`TypeRegistry`]. Does nothing if the entity does not
+    /// have a component of the same type, if the [`TypeRegistry`] does not contain the reflection data
+    /// for the given component, or if the entity does not exist.
```

Comment:
Nit: This also needs a line break
```suggestion
    /// Removes the component of the same type as the supplied boxed reflect component from the entity
    /// using the reflection data in the supplied [`TypeRegistry`]. 
    ///
    /// Does nothing if the entity does not have a component of the same type,
    /// if the [`TypeRegistry`] does not contain the reflection data for the given component,
    /// or if the entity does not exist.
```